src/plugins/vis_builder/public/application/components/data_tab/field_selector.tsx

```diff
@@ -65,6 +93,63 @@ export const FieldSelector = () => {
     [filteredFields]
   );
 
+  useEffect(() => {
+    async function getData() {
```

Comment:
There's no benefit (and a performance cost) to fetching the sampled hit data on every field click. The way this currently works (and discover works), is that you make a single fetch to get all the hit data, and then you can just locally calculate the field details on click. If we refactor in the future so that each field just makes a specific query via the agg service, yeah, there would be no need for a single fetch of hit data with all fields (and we could probably eliminate sampling).

But yeah a follow-up issue to do the aggregation via the agg service instead of client-side is definitely worth doing.

Reply:
I at least moved it all into a custom hook - I wanted to call that from the `Field` component, but I wasn't clever enough to figure out how to make sure we don't re-fetch the data needlessly that way, so it's still only called from `Field` selector. Not yet ideal, but cleaner for now until we can switch to the agg service instead of a single hits fetch.